gitpod-io/gitpod
# [server, dashboard] Do basic rate limiting on startWorkspace

## Description
This configures a rate-limit for API calls to `startWorkspace` of `1` / `10s`. The dashboard handles the error and waits until `retryAfter` to re-try the call (currently indefinitely).

Note that this is enforced per `server` instance at the moment, as the rate-limiter state is not shared between server instances. But it's a first step that already reduces the impact of future incidents.

## Related Issue(s)

Context: #8043

## How to test
 - start any workspace: https://gpl-8043-redirect-loop.staging.gitpod-dev.com/#https://github.com/gitpod-io/template-typescript-node
 - open the dev console to notice the rate limier messages

## Release Notes

```release-note
configure basic rate-limiting for `startWorkspace`
```

## Documentation
